### PR TITLE
[Admin-REST API] API to get role-scope mapping.

### PR DESCRIPTION
**Pull request by Meruja**
Fix https://github.com/wso2/product-apim/issues/8201 (https://github.com/wso2/product-apim/issues/7954)

This PR contains the GET operation only

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SettingsApiServiceImpl.java [hunk 1]
```diff
@@ -92,4 +93,23 @@ public Response settingsScopesScopeGet(String username, String scopeName, Messag
...
+     * @return Role-Scope list
+     */
+    @Override
+    public Response settingsScopesGet(MessageContext messageContext) {
+        try {
+            Map<String, String> roleScopeMapping = APIUtil.getRESTAPIScopesForTenant(MultitenantUtils
+                    .getTenantDomain(RestApiUtil.getLoggedInUsername()));
+            ScopeListDTO scopeListDTO = SettingsMappingUtil.fromScopeListToScopeListDTO(roleScopeMapping);
+            return Response.ok().entity(scopeListDTO).build();
+        } catch (Exception e) {
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SettingsApiServiceImpl.java:109]: Shall we catch the specific Exception here?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/SettingsMappingUtil.java [hunk 2]
```diff
@@ -136,4 +137,33 @@ private static SettingsKeyManagerConfigurationDTO fromKeyManagerConfigurationToS
...
+     * Converts api role-scope mapping to RoleScopeDTO List.
+     *
+     * @param roleScopeMapping Map of a Role Scope  Mapping
+     * @return RoleScopeDTO list
+     */
+    private static List<ScopeDTO>  fromRoleScopeMapToRoleScopeDTOList(Map<String, String>  roleScopeMapping) {
+        List<ScopeDTO> scopeDTOs = new ArrayList<>(roleScopeMapping.size());
+        for (Map.Entry<String, String>  mapping : roleScopeMapping.entrySet()) {
+            ScopeDTO roleScopeDTO = new ScopeDTO();
+                roleScopeDTO.setScope(mapping.getKey());
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/SettingsMappingUtil.java:163]: Shall we format this line?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SettingsApiServiceImpl.java (hunk 1 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SettingsApiServiceImpl.java:109]: PR has been reverted

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/SettingsMappingUtil.java (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/SettingsMappingUtil.java:163]: PR has been reverted

**Pull request merged**